src/components/GlobalPreferences/Notifications/ManageNotifications.js

```diff
@@ -0,0 +1,384 @@
+    )
+    const [selectedOrganization, setSelectedOrganization] = useState(-1)
+    const onOrganizationChange = useCallback(
+      idx => {
+        setSelectedOrganization(idx)
+      },
+      [setSelectedOrganization]
+    )
+
+    // Get unique app names by matching subscriptions with
+    const subscriptionApps = Array.from(
+      new Set(subscriptions.map(subscription => subscription.appName))
+    )
+    const [selectedApp, setSelectedApp] = useState(-1)
+    const onAppChange = useCallback(
+      idx => {
+        setSelectedApp(idx)
+      },
+      [setSelectedApp]
+    )
+    const events = Array.from(
+      new Set(subscriptions.map(subscription => subscription.eventName))
+    )
+    const [selectedEvent, setSelectedEvent] = useState(-1)
+    const onEventChange = useCallback(
+      idx => {
+        setSelectedEvent(idx)
+      },
+      [setSelectedEvent]
+    )
+    const onClearFilters = useCallback(() => {
+      setSelectedEvent(-1)
+      setSelectedApp(-1)
+      setSelectedOrganization(-1)
+    }, [setSelectedEvent, setSelectedApp, setSelectedOrganization])
+
+    const filteredSubscriptions = filterSubscriptions({
+      subscriptions,
+      event: events[selectedEvent],
+      appName: subscriptionApps[selectedApp],
+      organization: organizations[selectedOrganization],
+    })
+    const theme = useTheme()
+    return (
+      <DataView
+        fields={[
+          {
+            label: 'Organization',
+            priority: 3,
+          },
+          {
+            label: 'App',
+            priority: 2,
+          },
+          {
+            label: 'Contract',
+            priority: 2,
+          },
+          {
+            label: 'Event',
+            priority: 1,
+          },
+        ]}
+        heading={
+          <React.Fragment>
+            <div
+              css={`
+                height: ${9 * GU}px;
+                display: flex;
+                align-items: center;
+                justify-content: space-between;
+              `}
+            >
+              <div
+                css={`
+                  color: ${theme.content};
+                  ${textStyle('body1')}
+                `}
+              >
+                Subscriptions
+              </div>
+              {selectedSubscriptions.length > 0 && (
+                <div css="text-align: right;">
+                  <Button disabled={false} onClick={handleUnsubscribe}>
+                    {isSubmitting ? (
+                      <LoadingRing
+                        css={`
+                          margin-right: ${GU}px;
+                        `}
+                      />
+                    ) : (
+                      <IconMail
+                        css={`
+                          color: ${theme.negative};
+                          margin-right: ${GU}px;
+                        `}
+                      />
+                    )}{' '}
+                    Unsubscribe
```

Comment:
This one is hard and odd: when unsubscribing to an event from an app that was the first index of the dropdown and the user had previously subscribed to all events, when doing so, it pushes the app as the first index again and if that index was selected it modifies the selected app incorrectly:
https://www.loom.com/share/efc63a3d2a5e4f028c7663b740372878

Reply:
Fixed in https://github.com/aragon/aragon/pull/904/commits/c640eeb4f12d8e7a4e1b88e3603dbed16d8e0d18